siaskynet/skynet.py

```diff
@@ -76,10 +76,10 @@ def upload_directory_request(path, opts=None):
             opts = Skynet.default_upload_options()
 
         ftuples = []
-        files = list(Skynet.walk_directory(path).keys())
+        files = list(Skynet.walk_directory(path, path).keys())
         for file in files:
             ftuples.append((opts.portal_directory_file_fieldname,
-                            (file, open(file, 'rb'))))
+                            (file, open(path + file, 'rb'))))
```

Comment:
Could we have just done the following? 
```
 ftuples.append((opts.portal_directory_file_fieldname, (file[len(path):], open(file, 'rb'))))
```

Reply:
Yes, great suggestion. I've made the change.